### DISCUSSION TITLE
Hermit error

### DISCUSSION THREAD

**Discussion by Holden-VanKeyes** [cashapp/hermit#381] [Q&A]
Hello,
new to the Hermit world! setup was a breeze for me (Mac user), but trying to help my teammate (Windows) get setup and running into this error. I'm not sure I understand why this command would only error for her on Windows but works fine for me. I'm sure it's something simple, any guidance would be appreciated!
Hermit: error: expected one of "init" "version" "shell-hooks" etc...

`oras pull -o
Usage: hermit <command>
% Hermit is a hermetic binary
hermit: error: expected one of "init".
"version". "manifest".
"info".
"shell-hooks'`

**Comment by alecthomas** [comment]
It's hard to tell what you're doing without more context, but I suspect you are not activating your Hermit environment: `. ./bin/activate-hermit`

**Comment by Holden-VanKeyes** [comment]
thanks Alec! Hermit environment appears activated
![Screen Shot 2023-06-14 at 4 43 42 PM](https://github.com/cashapp/hermit/assets/102389369/1be0b982-deb4-4e1f-a531-7281c1c2af01)
this Make command runs an oras pull script from git repo - with Hermit deactivated this errors out as 'no command' found as expected, but in Hermit we get the previously mentioned Hermit error. Any other context I can provide?

**Reply by alecthomas** [reply]
You'll need to track down which _exact_ command is triggering the error. Again, there's not enough context in that screenshot to give any useful feedback.

**Reply by Holden-VanKeyes** [reply]
So the exact command is the one I mentioned:
`oras pull -o data registry.gitlab.com/boaterslist/frontend/`
but I'd be happy to provide more context. Is there something specific?

**Reply by alecthomas** [reply]
And if you type that at the command-line, with the environment activated, it fails?

eg.

```
$ oras pull -o data registry.gitlab.com/boaterslist/frontend/
```

**Reply by alecthomas** [reply]
Also can you paste the output of `./bin/hermit --version` please?

**Reply by alecthomas** [reply]
And with the environment activated, the output of `which oras` and `echo $PATH`.

**Reply by Holden-VanKeyes** [reply]
v0.36.0 (stable)

**Reply by Holden-VanKeyes** [reply]
apologies, she's gone home for the day, but I do have a previous screenshot where I asked her to run hermit env which includes her Path
![image](https://github.com/cashapp/hermit/assets/102389369/9a64268a-46d3-4960-8b30-7f75da300d8e)

**Reply by alecthomas** [reply]
What I'd suggest trying, in order, are:

1. Fresh clone of the repository then try the same commands to see if they succeed
2. If that fails, run `hermit info hermit --json | jq -r '.[].Root' | xargs dirname | xargs dirname` and delete that entire directory (it should be something like `/Users/alec/Library/Caches/hermit` on Mac).
 3. If the above fails, repeat 2 but in addition, do this: `rm ~/bin/hermit*`

**Reply by Holden-VanKeyes** [reply]
ah, I see, they're on a Windows machine... is hermit Mac only? Otherwise, I'll try all these steps with her tomorrow. Appreciate it Alec!

**Reply by alecthomas** [reply]
Hermit supports Mac and Linux, but not Windows unfortunately - not because it can't, but we just don't use Windows.
